File: src/main/java/com/soapdogg/lispInterpreter/functions/LispFunction.java
package com.soapdogg.lispInterpreter.functions;

import com.soapdogg.lispInterpreter.datamodels.Node;
import com.soapdogg.lispInterpreter.datamodels.UserDefinedFunction;

import java.util.List;
import java.util.Map;

public interface LispFunction {
    Node evaluateLispFunction(
        final Node params,
        List<UserDefinedFunction> userDefinedFunctions,
        Map<String, Node> variableNameToValueMap
    ) throws Exception;
}

File: src/main/java/com/soapdogg/lispInterpreter/functions/LispFunction.kt
package com.soapdogg.lispInterpreter.functions

import com.soapdogg.lispInterpreter.datamodels.Node
import com.soapdogg.lispInterpreter.datamodels.UserDefinedFunction

interface LispFunction {
    fun evaluateLispFunction(
        params: Node,
        userDefinedFunctions: List<UserDefinedFunction>,
        variableNameToValueMap: Map<String, Node>
    ): Node
}